server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -325,6 +354,15 @@ static long getPrimaryTerm(String[] filenameTokens) {
         static long getGeneration(String[] filenameTokens) {
             return RemoteStoreUtils.invertLong(filenameTokens[2]);
         }
+
+        public static Tuple<Tuple<Long, Long>, String> getNodeIdByPrimaryTermAndGeneration(String filename) {
```

Comment:
Does it simplify things if the return type is: `Map<String, List<String>>`? Key of the map is prefix of the metadata file that we need to consider for comparison and value would be list of nodeIds. If map contains a key with value list size > 1, we have a conflict.

Reply:
I don't have a strong opinion on this . `Tuple<Long,Long>` makes it easier to extract primary term and generation .  But whatever you are suggesting makes the code simpler . would implement it in next revision .